### DISCUSSION TITLE
Help - Did everything for NGINX exactly as instructed, yet nothing shows on the domain that I want to acess

### DISCUSSION THREAD

**Discussion by Frityet** [coder/code-server#2567] [Q&A] [answered]
going to the local ip of the machine that is running it does show the default NGINX start page

**Comment by Frityet** [comment]
alright, going to the local ip on port 8080 shows code-server, but the domain still dont work

**Comment by Frityet** [comment]
Now, it just shows the ngnix start page, if I have ngnix on

**Comment by code-asher** [comment]
Could you post your NGINX config and the command or config you are using
to start code-server?

**Reply by Frityet** [reply]
```yaml
server {
    listen 80;
    listen [::]:80;
    server_name [my domain];

    location / {
      proxy_pass http://localhost:8080/;
      proxy_set_header Host $host;
      proxy_set_header Upgrade $http_upgrade;
      proxy_set_header Connection upgrade;
      proxy_set_header Accept-Encoding gzip;
    }
}
```

I use `code-server` to start code server (for some reason it wont let me interact with shell unless I have cntrl-c after)

`sudo certbot --non-interactive --redirect --agree-tos --nginx -d [mydomain] -m [email]@gmail.com`
this is how I launch nginx

**Comment by code-asher** [comment]
Your NGINX config looks good!

Ctrl-c will kill the process so you might be killing code-server.

There are a few ways around this:
 - Use & to spawn in the background: `code-server & && disown`
 - Use a systemd service to spawn code-server

The second option might be a bit better because it will automatically
restart code-server if it crashes. If you use our install script a
service will be installed automatically. If not then you can look at ours to
create your own: https://github.com/cdr/code-server/blob/v3.8.0/ci/build/code-server@.service

The certbot command will set up TLS but I don't think it runs NGINX.
You'll need something like:

```
  systemctl enable nginx # This is so it starts automatically on reboot.
  systemctl start nginx  # This starts NGINX.
```

**Reply by Frityet** [reply]
thank you! I still am having the issue of not being able to acess code-server not on my local network

**Reply by Frityet** [reply]
doing 
```bash
code-server & && disown
``` 
gives me error about the &&

**Reply by Frityet** [reply]
by just doing one `&` i get the same issue

**Reply by Frityet** [reply]
nevermind it worked with one `&`

**Comment by Frityet** [comment]
![Uploading Screen Shot 2021-01-12 at 11.55.36 AM.png…]()

Here is my domain DNS settings

**Comment by Frityet** [comment]
Update: typing in my public ip + :8080 works perfectly, however domain still not working, here is a FIXED picture of my dns settings for my domain
<img width="1022" alt="Screen Shot 2021-01-12 at 11 55 36 AM" src="https://user-images.githubusercontent.com/25426233/104395257-335b1b80-54fd-11eb-9c66-539b0dd1ff77.png">

**Reply by code-asher** [reply]
Cool, the DNS settings look good to me. Did you forward ports 80 and 443? And your NGINX config has the sub-domain in it, right? e.g. `server_name code.<domain>`.

**Reply by Frityet** [reply]
> Cool, the DNS settings look good to me. Did you forward ports 80 and 443? And your NGINX config has the sub-domain in it, right? e.g. `server_name code.<domain>`.

I did not forward 80 and 433 (whoops) and by ngix config do you mean the code-server site?

**Reply by code-asher** [reply]
There should be a part that looks like this:
```
server {
    listen 80;
    listen [::]:80;
    server_name code.mydomain.com

    location / {
      proxy_pass http://localhost:8080/;
      proxy_set_header Host $host;
      proxy_set_header Upgrade $http_upgrade;
      proxy_set_header Connection upgrade;
      proxy_set_header Accept-Encoding gzip;
    }
}
```

Also make sure it's in `/etc/nginx/sites-enabled` (you can put it there directly or put it in `/etc/nginx/site-available` and symlink it into `sites-enabled`, that's the standard way to do it).

**Reply by Frityet** [reply]
> There should be a part that looks like this:
> 
> ```
> server {
>     listen 80;
>     listen [::]:80;
>     server_name code.mydomain.com
> 
>     location / {
>       proxy_pass http://localhost:8080/;
>       proxy_set_header Host $host;
>       proxy_set_header Upgrade $http_upgrade;
>       proxy_set_header Connection upgrade;
>       proxy_set_header Accept-Encoding gzip;
>     }
> }
> ```
> 
> Also make sure it's in `/etc/nginx/sites-enabled` (you can put it there directly or put it in `/etc/nginx/site-available` and symlink it into `sites-enabled`, that's the standard way to do it).

yeah, it says code.mydomain.com, and its in sites enabled

**Reply by code-asher** [reply]
Hmm in that case it seems like you have everything properly configured! You might need to check the nginx logs when you go to `http://code.domain.com` and see if there are any errors.

You might also want to check out `--link` which will set up all the proxying for you automatically by going through our cloud.

**Reply by Frityet** [reply]
> Hmm in that case it seems like you have everything properly configured! You might need to check the nginx logs when you go to `http://code.domain.com` and see if there are any errors.
> 
> You might also want to check out `--link` which will set up all the proxying for you automatically by going through our cloud.

Does --link require any account or payment? I thought it did thats why I havent used it

**Reply by code-asher** [reply]
It looks like you've already tried it out but just to confirm: nope, it's completely free and authenticates using your GitHub account.

**Comment by Frityet** [comment]
```properties
user www-data;
worker_processes auto;
pid /run/nginx.pid;
include /etc/nginx/modules-enabled/*.conf;

events {
	worker_connections 768;
	# multi_accept on;
}

http {

	##
	# Basic Settings
	##

	sendfile on;
	tcp_nopush on;
	tcp_nodelay on;
	keepalive_timeout 65;
	types_hash_max_size 2048;
	# server_tokens off;

	# server_names_hash_bucket_size 64;
	# server_name_in_redirect off;

	include /etc/nginx/mime.types;
	default_type application/octet-stream;

	##
	# SSL Settings
	##

	ssl_protocols TLSv1 TLSv1.1 TLSv1.2 TLSv1.3; # Dropping SSLv3, ref: POODLE
	ssl_prefer_server_ciphers on;

	##
	# Logging Settings
	##

	access_log /var/log/nginx/access.log;
	error_log /var/log/nginx/error.log;

	##
	# Gzip Settings
	##

	gzip on;

	# gzip_vary on;
	# gzip_proxied any;
	# gzip_comp_level 6;
	# gzip_buffers 16 8k;
	# gzip_http_version 1.1;
	# gzip_types text/plain text/css application/json application/javascript text/xml application/xml application/xml+rss text/javascript;

	##
	# Virtual Host Configs
	##

	include /etc/nginx/conf.d/*.conf;
	include /etc/nginx/sites-enabled/*;
}


#mail {
#	# See sample authentication script at:
#	# http://wiki.nginx.org/ImapAuthenticateWithApachePhpScript
# 
#	# auth_http localhost/auth.php;
#	# pop3_capabilities "TOP" "USER";
#	# imap_capabilities "IMAP4rev1" "UIDPLUS";
# 
#	server {
#		listen     localhost:110;
#		protocol   pop3;
#		proxy      on;
#	}
# 
#	server {
#		listen     localhost:143;
#		protocol   imap;
#		proxy      on;
#	}
#}
```

Here is my nginx config

**Comment by Frityet** [comment]
I port forwarded 80 and 433, and still nothing

**Comment by Frityet** [comment, marked as answer]
Worked when doing `code-server --link` Thank you so much asher!

**Reply by code-asher** [reply]
Woohoo I'm elated to hear that! Happy programming!

**Reply by geenva** [reply]
Confirming this worked. Thanks to Frityet and code-asher!
